lib/src/ui/home_page.dart

```diff
@@ -0,0 +1,127 @@
+    required this.title,
+  });
+
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  void initState() {
+    super.initState();
+    widget.iMoviesBloc.fetchMoviesApi();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        centerTitle: true,
+        title: Text(
+          widget.title,
+        ),
+        leading: IconButton(
+          icon: Icon(
+            Icons.menu,
+          ),
+          onPressed: () {},
+        ),
+        actions: [
+          IconButton(
+            icon: Icon(
+              Icons.notifications_none,
+            ),
+            onPressed: () {},
+          ),
+          IconButton(
+            icon: Icon(
+              Icons.search,
+            ),
+            onPressed: () {},
+          ),
+        ],
+        flexibleSpace: Container(
+          decoration: BoxDecoration(
+            gradient: LinearGradient(
+              colors: [
+                Colors.red,
+                Colors.black,
+              ],
+              begin: Alignment.bottomRight,
+              end: Alignment.topLeft,
+            ),
+          ),
+        ),
+        elevation: Dimension.shadowAppBar,
+      ),
+      body: StreamBuilder(
+        stream: widget.iMoviesBloc.streamMovies,
+        builder: (
+          context,
+          AsyncSnapshot<MovieModel> snapshot,
+        ) {
+          return snapshot.hasData
+              ? buildList(
+                  snapshot,
+                )
+              : Center(
+                  child: CircularProgressIndicator(
+                    backgroundColor: Colors.red,
+                    color: Colors.lightGreenAccent,
+                  ),
+                );
+        },
+      ),
+    );
+  }
+
+  Widget buildList(AsyncSnapshot<MovieModel> snapshot) {
+    return GridView.builder(
+      itemCount: snapshot.data!.results.length,
+      gridDelegate: new SliverGridDelegateWithFixedCrossAxisCount(
+        crossAxisCount: Dimension.gridColumns,
+      ),
+      itemBuilder: (
+        BuildContext context,
+        int index,
+      ) {
+        return GridTile(
+          child: Container(
+            margin: EdgeInsets.all(
+              Dimension.gridTileMargin,
+            ),
+            decoration: BoxDecoration(
+              boxShadow: [
+                BoxShadow(
+                  color: Colors.black.withOpacity(
+                    Dimension.boxShadowOpacity,
+                  ),
+                  spreadRadius: Dimension.boxShadowSpreadRadius,
+                  blurRadius: Dimension.boxShadowBlurRadius,
+                ),
+              ],
+            ),
+            child: Image.network(
+              StringConstant.imagesPath +
+                  snapshot.data!.results[index].posterPath,
+              fit: BoxFit.fill,
+            ),
+          ),
+        );
+      },
+    );
+  }
```

Comment:
Instead to do this, you can add a ternary operator like:
```suggestion
  Widget buildList(AsyncSnapshot snapshot) {
    return snapshot.hasData
        ? GridView.builder(
            itemCount: snapshot.data.results.length,
            gridDelegate: new SliverGridDelegateWithFixedCrossAxisCount(
              crossAxisCount: Dimension.gridColumns,
            ),
            itemBuilder: (
              BuildContext context,
              int index,
            ) {
              return GridTile(
                child: Container(
                  margin: EdgeInsets.all(
                    Dimension.gridTileMargin,
                  ),
                  decoration: BoxDecoration(
                    boxShadow: [
                      BoxShadow(
                        color: Colors.black.withOpacity(
                          Dimension.boxShadowOpacity,
                        ),
                        spreadRadius: Dimension.boxShadowSpreadRadius,
                        blurRadius: Dimension.boxShadowBlurRadius,
                      ),
                    ],
                  ),
                  child: Image.network(
                    StringConstant.imagesPath + snapshot.data!.results[index].posterPath,
                    fit: BoxFit.fill,
                  ),
                ),
              );
            },
          )
        : Container();
  }
```

Reply:
done, thank you